src/pools/D3MAaveDaiPool.sol

```diff
@@ -0,0 +1,178 @@
+        require((z = x + y) >= x, "D3MAaveDaiPool/overflow");
+    }
+    function _sub(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require((z = x - y) <= x, "D3MAaveDaiPool/underflow");
+    }
+    function _mul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        require(y == 0 || (z = x * y) / y == x, "D3MAaveDaiPool/overflow");
+    }
+    uint256 constant RAY  = 10 ** 27;
+    function _rmul(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, y) / RAY;
+    }
+    function _rdiv(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _mul(x, RAY) / y;
+    }
+    function _min(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = x <= y ? x : y;
+    }
+
+    // --- Admin ---
+    function file(bytes32 what, address data) public override auth {
+        require(live == 1, "D3MAaveDaiPoolPool/no-file-not-live");
+
+        if (what == "king") king = data;
+        else super.file(what, data);
+    }
+
+    function validTarget() external view override returns (bool) {
+        (,,,,,,,,,, address strategy,) = LendingPoolLike(pool).getReserveData(address(asset));
+        return strategy == address(interestStrategy);
+    }
+
+    function calcSupplies(uint256 availableAssets) external view override returns(uint256, uint256) {
+        return D3MPlanLike(plan).calcSupplies(availableAssets);
+    }
+
+    // Deposits Dai to Aave in exchange for adai which gets sent to the msg.sender
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#deposit
+    function deposit(uint256 amt) external override auth {
+        LendingPoolLike(pool).deposit(address(asset), amt, address(this), 0);
+
+    }
+
+    // Withdraws Dai from Aave in exchange for adai
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#withdraw
+    function withdraw(uint256 amt) external override auth {
+        LendingPoolLike(pool).withdraw(address(asset), amt, address(this));
+    }
+
+    // --- Collect any rewards ---
+    function collect(address[] memory assets, uint256 amount) external override auth returns (uint256 amt) {
+        require(king != address(0), "D3MAaveDaiPool/king-not-set");
+
+        amt = rewardsClaimer.claimRewards(assets, amount, king);
+    }
+
+    function transferShares(address dst, uint256 amt) external override returns(bool) {
+        return ShareTokenLike(share).transfer(dst, amt);
+    }
+
+    // --- Balance in standard ERC-20 denominations
+    function assetBalance() external view override returns (uint256) {
+        return ShareTokenLike(share).scaledBalanceOf(address(this));
+    }
+
+    function shareBalance() public view override returns(uint256) {
+        return ShareTokenLike(share).balanceOf(address(this));
+    }
+
+    function maxWithdraw() external view override returns(uint256) {
+        // return TokenLike(asset).balanceOf(share);
```

Comment:
Can this be uncommented, or does it need a TODO?

Reply:
todos, thanks